packages/contract-wrappers/src/contract_wrappers/erc721_token_wrapper.ts

```diff
@@ -0,0 +1,439 @@
+        const isProxyApproved = approvedAddress === proxyAddress;
+        return isProxyApproved;
+    }
+    /**
+     * Enable or disable approval for a third party ("operator") to manage all of `ownerAddress`'s assets.
+     * Throws if `_tokenId` is not a valid NFT
+     * Emits the ApprovalForAll event.
+     * @param   tokenAddress    The hex encoded contract Ethereum address where the ERC721 token is deployed.
+     * @param   ownerAddress    The hex encoded user Ethereum address of the token owner.
+     * @param   operatorAddress The hex encoded user Ethereum address of the operator you'd like to set approval for.
+     * @param   isApproved      The boolean variable to set the approval to.
+     * @param   txOpts          Transaction parameters.
+     * @return  Transaction hash.
+     */
+    public async setApprovalForAllAsync(
+        tokenAddress: string,
+        ownerAddress: string,
+        operatorAddress: string,
+        isApproved: boolean,
+        txOpts: TransactionOpts = {},
+    ): Promise<string> {
+        assert.isETHAddressHex('tokenAddress', tokenAddress);
+        assert.isETHAddressHex('ownerAddress', ownerAddress);
+        assert.isETHAddressHex('operatorAddress', operatorAddress);
+        assert.isBoolean('isApproved', isApproved);
+        const normalizedTokenAddress = tokenAddress.toLowerCase();
+        const normalizedOwnerAddress = ownerAddress.toLowerCase();
+        const normalizedOperatorAddress = operatorAddress.toLowerCase();
+
+        const tokenContract = await this._getTokenContractAsync(normalizedTokenAddress);
+        const txHash = await tokenContract.setApprovalForAll.sendTransactionAsync(
+            normalizedOperatorAddress,
+            isApproved,
+            {
+                gas: txOpts.gasLimit,
+                gasPrice: txOpts.gasPrice,
+                from: normalizedOwnerAddress,
+            },
+        );
+        return txHash;
+    }
+    /**
+     * Enable or disable approval for a third party ("operator") to manage all of `ownerAddress`'s assets.
+     * Throws if `_tokenId` is not a valid NFT
+     * Emits the ApprovalForAll event.
+     * @param   tokenAddress    The hex encoded contract Ethereum address where the ERC721 token is deployed.
+     * @param   ownerAddress    The hex encoded user Ethereum address of the token owner.
+     * @param   operatorAddress The hex encoded user Ethereum address of the operator you'd like to set approval for.
+     * @param   isApproved      The boolean variable to set the approval to.
+     * @param   txOpts          Transaction parameters.
+     * @return  Transaction hash.
+     */
+    public async setProxyApprovalForAllAsync(
+        tokenAddress: string,
+        ownerAddress: string,
+        isApproved: boolean,
+        txOpts: TransactionOpts = {},
+    ): Promise<string> {
+        const proxyAddress = this._erc721ProxyWrapper.getContractAddress();
```

Comment:
Missing assertions.

Reply:
Nope. We call our method here and it has assertions so they're redundant